torchreid/utils/tools.py

```diff
@@ -16,16 +16,24 @@
 import random
 import sys
 import time
+import subprocess
 
 import numpy as np
 import torch
 import cv2 as cv
 
 __all__ = [
     'mkdir_if_missing', 'check_isfile', 'set_random_seed', "worker_init_fn",
-    'read_image', 'get_model_attr', 'StateCacher', 'random_image', 'EvalModeSetter'
+    'read_image', 'get_model_attr', 'StateCacher', 'random_image', 'EvalModeSetter', 'get_git_revision'
 ]
 
+def get_git_revision():
+    path = os.getcwd()
```

Comment:
os.getcwd() returns current directory, but it could be parent to d-o-r repo.
Actually I've meant
```python
path = os.path.abspath(os.path.dirname(__file__))
sha_message = ['git', 'rev-parse', 'HEAD']
subprocess.check_output(sha_message, cwd=path).decode('ascii').strip()
```

Reply:
please, check.
I've checked that this works from another folder